client/src/components/HeaderSearch/index.tsx

```diff
@@ -27,20 +27,17 @@ const HeaderSearch = () => {
         setStatusMsg(`Sorry, your entered location "${inputValue}" does not exists`);
         setInputValue('');
       } else {
-        const coords = {
-          lat: locationData.locations[0].lat,
-          lon: locationData.locations[0].lon
-        };
-        setCoordinates(coords);
+        setCoordinates(null);
         setStatusMsg(null);
-        navigate(`/details/${locationData.locations[0].name}`);
         setInputValue('');
+        navigate(`/details/${locationData.locations[0].name.replace(/\s+/g, '')}`);
```

Comment:
please move `locationData.locations[0].name.replace(/\s+/g, '')` to separate variable with clear variable name that could explain what way location name was formatted